torch/fx/experimental/symbolic_shapes.py

```diff
@@ -6,6 +6,32 @@
 except ImportError:
     HAS_SYMPY = False
 
+aten = torch.ops.aten
+
+def create_contiguous(shape):
+    strides = [1]
+    for dim in reversed(shape[:-1]):
+        strides.append(dim * strides[-1])
+    return list(reversed(strides))
+
+
+def is_symbolic_op(func):
+    return func in [aten.sym_size.default, aten.dim.default, aten.is_contiguous.default, aten.stride]
+
+
+def handle_symbolic_op(func, args, kwargs):
+    assert is_symbolic_op(func)
+    if func == torch.ops.aten.sym_size.default:
+        return None
+    if func == torch.ops.aten.dim.default:
+        return len(args[0].shape)
+    # TODO: hack, need to make is_contiguous calls symbolic (probably through symbolic strides)
```

Comment:
We don't have symbolic bools, it probably will be easiest to specialize on contiguity and have that generate a number of constraints based on it being contiguous. https://github.com/albanD/subclass_zoo/blob/main/dynamic_strides.ipynb

Reply:
yeah, we can't do that today though, as checking contiguity depends on the strides.